python/cudf/cudf/utils/dtypes.py

```diff
@@ -643,7 +643,12 @@ def find_common_type(dtypes):
         dtypes = dtypes - td_dtypes
         dtypes.add(np.result_type(*td_dtypes))
 
-    return np.find_common_type(list(dtypes), [])
+    common_dtype = np.find_common_type(list(dtypes), [])
+    if common_dtype == np.dtype("float16"):
+        # cuDF does not support float16 dtype
+        return np.dtype("float32")
+    else:
+        return common_dtype
```

Comment:
Not sure we want to tackle it in this PR, but this won't handle Decimal Types.

Reply:
Raising an error here incase of decimal types..